tests/test_crf.py

```diff
@@ -476,3 +476,134 @@ def assert_close(actual, expected):
+                                             f"not matching non-scripted forward output {llh}"
+
+        # Test scripted forward when running batch first mode
+        crf_bf = make_crf(batch_first=True)
+        # Copy parameter values from non-batch-first CRF; requires_grad must be False
+        # to avoid runtime error of in-place operation on a leaf variable
+        crf_bf.start_transitions.requires_grad_(False).copy_(crf.start_transitions)
+        crf_bf.end_transitions.requires_grad_(False).copy_(crf.end_transitions)
+        crf_bf.transitions.requires_grad_(False).copy_(crf.transitions)
+        crf_bf_script = torch.jit.script(crf_bf)
+        emissions = emissions.transpose(0, 1)
+        # shape: (batch_size, seq_length)
+        tags = tags.transpose(0, 1)
+        llh_bf = crf_bf(emissions, tags)
+        llh_bf_script = crf_bf_script(emissions, tags)
+        assert torch.equal(llh_bf_script, llh_bf), f"scripted crf forward output {llh_bf_script} " \
+                                                   f"not matching non-scripted forward output {llh_bf}"
+
+    def test_scripted_decode(self):
+        # Test decoding with a mask
+        crf = make_crf()
+        crf_script = torch.jit.script(crf)
+
+        seq_length, batch_size = 3, 2
+        # shape: (seq_length, batch_size, num_tags)
+        emissions = make_emissions(crf, seq_length, batch_size)
+        # mask should be (seq_length, batch_size)
+        mask = torch.tensor([[1, 1, 1], [1, 1, 0]], dtype=torch.uint8).transpose(0, 1)
+        best_tags = crf.decode(emissions, mask=mask)
+        best_tags_scripted = crf_script.decode(emissions, mask=mask)
+        assert best_tags == best_tags_scripted, f"scripted decode output {best_tags_scripted} " \
+                                                f"doesn't match non-scripted output {best_tags}"
+
+        # Test decoding without a mask
+        best_tags_no_mask = crf.decode(emissions)
+        best_tags_no_mask_scripted = crf_script.decode(emissions)
+        assert best_tags_no_mask == best_tags_no_mask_scripted, f"scripted decode output {best_tags_no_mask_scripted} " \
+                                                                f"doesn't match non-scripted output {best_tags_no_mask}"
+
+        # Test batched decode
+        batch_size, seq_length = 2, 3
+        # shape: (seq_length, batch_size, num_tags)
+        emissions_batched = make_emissions(crf, seq_length, batch_size)
+        # shape: (seq_length, batch_size)
+        mask_batched = torch.tensor([[1, 1, 1], [1, 1, 0]], dtype=torch.uint8).transpose(0, 1)
+        batched = crf.decode(emissions_batched, mask=mask_batched)
+        batched_scripted = crf_script.decode(emissions_batched, mask=mask_batched)
+        assert batched == batched_scripted, f"scripted decode output {batched_scripted} " \
+                                            f"doesn't match non-scripted output {batched}"
```

Comment:
This seems a duplicate of lines 569-577.

Reply:
It is duplicated, so I removed it.